node/src/components/consensus/validator_change.rs

```diff
@@ -0,0 +1,50 @@
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+
+use casper_types::PublicKey;
+
+use super::era_supervisor::Era;
+
+/// A change to a validator's status between two eras.
+#[derive(Serialize, Deserialize, Debug, JsonSchema)]
+pub enum ValidatorChange {
+    /// The validator got newly added to the validator set.
+    Added,
+    /// The validator was removed from the validator set.
+    Removed,
+    /// The validator was banned from this era.
+    Banned,
+    /// The validator was excluded from proposing new blocks in this era.
+    CannotPropose,
+    /// We saw the validator misbehave in this era.
+    SeenAsFaulty,
+}
+
+impl ValidatorChange {
+    pub(super) fn era_changes<I>(
```

Comment:
There's some `if`-based logic hidden in this function, so I think it's a good candidate for a unit test.

Reply:
So this code isn't even mine, its part of previously reviewed PR that has been merged, and I am not too keen on changing it.
The PR: https://github.com/casper-network/casper-node/pull/1540

Reply:
I'm not suggesting changes to the code, but just covering it with test.

Reply:
So you're suggesting a test to catch a mistaken inversion in the `if` logic, is that correct??

Reply:
Yes, exactly that.

Reply:
Added tests